test/functional/specs/C28754.js

```diff
@@ -0,0 +1,41 @@
+import { t } from "testcafe";
+import createNetworkLogger from "../helpers/networkLogger";
+import { responseStatus } from "../helpers/assertions";
+import fixtureFactory from "../helpers/fixtureFactory";
+import baseConfig from "../helpers/constants/baseConfig";
+import configureAlloyInstance from "../helpers/configureAlloyInstance";
+
+const networkLogger = createNetworkLogger();
+
+fixtureFactory({
+  title:
+    "C28754 - Consenting to no purposes should result in no data handles in the response.",
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+
+test.meta({
+  ID: "C28754",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+test("C28754 - Consenting to no purposes should result in no data handles in the response.", async () => {
+  await configureAlloyInstance("alloy", {
+    defaultConsent: { general: "pending" },
+    ...baseConfig
+  });
+
+  await t.eval(() => window.alloy("setConsent", { general: "out" }));
+
+  await responseStatus(networkLogger.edgeEndpointLogs.requests, 200);
+  const errorMessage = await t.eval(() =>
+    window
+      .alloy("event", { data: { a: 1 } })
+      .then(() => undefined, e => e.message)
+  );
+
+  await t.expect(networkLogger.edgeEndpointLogs.requests.length).eql(0);
+
+  await t.expect(errorMessage).ok("Expected the event command to be rejected");
+  await t.expect(errorMessage).contains("user declined consent");
```

Comment:
This needs more details added to make sure we are testing the use case:

- Retrieve the response JSON or converted into an Alloy response object
- Make sure it does not contain ID syncs handles
- Make sure it does not contain audiences
- Make sure it does not contain personalization

Reply:
Misunderstood the expected results, making the changes now.

Reply:
Check out C14394, how `createResponse` is being used.